Example Christian Religious Changes

LT-Rascek opened the pull request:
First cut on importing religions from WTWSMS to CK3. New tenets added to support some of the WTWSMS Christian religions (e.g., Donatism, Marcionism, Montanism, &c.).

Of particular interest is the creation of "Religion Doctrines" for specific religions (only Christianity at this time). This will only show up for Christian religions and relate to specific questions like Christology, Pneumatology, and Sotierology that are specific to Christian religions. This, I hope will allow consolidation of some of the older WTWSMS religions like Pelagianism, Antinomianism, and Total Depravity into a more flexible, meaningful framework, since there's no real reason we can't have an antinomian Nicene heresy or a Pelagian Nestorian one.

Not yet included is a (currently very) crude event that can be used to change tenets & doctrines within faiths (rather than spawning new faiths) to help better deal with issues like the Schism of the Three Chapters, Nestorian Synod of Beth Lapat, and the like; the event successfully changed Catholicism to replace Monasticism with Islamic Syncretism and changed the Christology to Miaphysite. 

Needs polish on this stuff, but implementing it should make Christianity here much more flexible and interesting than in ck2 WTWSMS.

Examples in practice:
![_christian_doctrine_changes_example](https://user-images.githubusercontent.com/55886300/92975222-ce5fbe00-f455-11ea-87ba-77127664c304.jpg)
Christian Doctrine Categories; trying to make a Monothelitic, Binitarian, Pnematomachi, Memorialist, Antinomian, Judaizing Christian sect that is exactly like Catholicism in all other respects.

![_christian_doctrine_example](https://user-images.githubusercontent.com/55886300/92975252-de779d80-f455-11ea-9fcd-0644b0adc76a.jpg)
Up close view of Dyophysite with localization

![_ecstatic_prophecy_example](https://user-images.githubusercontent.com/55886300/92975283-ea635f80-f455-11ea-8878-200d72644cfe.jpg)
Ecstatic Prophecy Tenet Overview

![_religion_arianism_example](https://user-images.githubusercontent.com/55886300/92975307-f51df480-f455-11ea-96e4-5ed7eb8cfb91.jpg)
Arianism Overview

![_tenet_iconoclasm_example](https://user-images.githubusercontent.com/55886300/92975319-fa7b3f00-f455-11ea-8b1a-b316a9527708.jpg)
Iconoclasm Overview

![_tenet_rigorism_example](https://user-images.githubusercontent.com/55886300/92975342-06670100-f456-11ea-857f-61c695859acc.jpg)
Donatism Rigorism Tenet Overview

![_muslim_no_christian_doctrines](https://user-images.githubusercontent.com/55886300/92975354-0d8e0f00-f456-11ea-9f1c-f6038d0ca5c4.jpg)
Muslims currently have no Religion doctrines

![_islamic_catholicism_example](https://user-images.githubusercontent.com/55886300/92975656-be94a980-f456-11ea-9265-f962c89436d2.jpg)
Crude event example of Islamic Catholicism.

LT-Rascek commented:
Also realized there's a localization conflict with BP_religion_core_tenets_l_english.yml so I'll need to fix that.

loup99 commented on BP/localization/english/religion/BP_religion_christianity_l_english.yml:
```diff
@@ -0,0 +1,147 @@
+ georgian_orthodox_adherent:0 "Kartvelian"
+ georgian_orthodox_adherent_plural:0 "Kartvelians"
+ georgian_orthodox_desc:0 "The Georgian Orthodox Church is the head church of the Georgian people, founded by the Kingdom of Iberia."
+ #Syrian Orthodox
```
This is a religion you decided to add, but I'm not sure a separate "Syriac Orthodox" branch is necessary, since the conflicts are already present through the Monophysites and Miaphysites.

LT-Rascek replied:
Al-Harith ibn Jabalah (528) was a major Ghassanid patron of the Syriac Orthodox Church, which is the only really specific information I've found on Ghassanid religiosity. I haven't found too many sources that give much detail about the actual details of the Monophysite faith of the Ghassanids; I'm suspecting that they may have nominally fallen under the Antiochene See, or one of the competing claimants for the See, given the confusion between Monophsyites and Miaphystism/Severians.

On the other hand, you might have resources that better outline the specifics in better detail. I can be replaced with a monophysite faith with very little trouble.

LT-Rascek replied:
For completeness:  I'll also note some specific Christian faiths from WTWSMS were not included on this pull request and instead modeled as Christian religious doctrines:
Adoptionist, Antinomian, Apollinarian,  Binitarian, Eutychian, Melchisedechian, Monarchian, Monophsyite, Monothelite, Pelagian, Pneumathomachian, Total Depravity, 
And some that weren't added don't that have specific doctrines/tenets associated with them:
Patripassian, Aquileian, Fraticelli, Psilantrhopist

loup99 replied:
> Al-Harith ibn Jabalah (528) was a major Ghassanid patron of the Syriac Orthodox Church, which is the only really specific information I've found on Ghassanid religiosity. I haven't found too many sources that give much detail about the actual details of the Monophysite faith of the Ghassanids; I'm suspecting that they may have nominally fallen under the Antiochene See, or one of the competing claimants for the See, given the confusion between Monophsyites and Miaphystism/Severians.

On the distinction between Miaphysites and Monophysites with regards to the Ghassanids:

> The Ghassanids are Monophysite in the _The Oxford Dictionary of Byzantium_, but Miaphysite in _The Oxford Dictionary of the Middle Ages_. In all precise references they are Monophysite but in more general descriptions Miaphysite.
> Why the confusion? Because outside the specific literature Miaphysite is being used as a more general term encompassing both the more moderate Miaphysites and the more radical Monophysites.
> By the way, "Miaphysite" does not exist as an entry in the Oxford Dictionaries, instead they speak of moderate and extreme monophysites.

> On the other hand, you might have resources that better outline the specifics in better detail. I can be replaced with a monophysite faith with very little trouble. With a little bit of event writing, the actual Christology of the Syriac Orthodox Branch can start as Monophysite and can change if they loose Ghassanid patronage.

To be clear, I do not think there should be a Syriac Orthodox religion at start. I think it is sufficient to have Nicene, Miaphysite and Monophysite faiths, which then represent the different christological tendencies, more moderate or more radical. The label "Syriac Orthodox" is more confusing than anything else in 476, since it describes a cultural distinction rather than a theological and doctrinal one. The later "Syriac" or "Jacobite" Church emerged from Jacob Baradaeus, which was not even born at start. Yet there were still Miaphysites and Monophysites before Baradaeus, since the one nature belief had been condemned by the Council of Chalcedon.

loup99 commented on BP/localization/english/religion/BP_religion_core_tenets_l_english.yml:
```diff
@@ -1,6 +1,38 @@
- tenet_kharis_name:0 "Kharis"
- tenet_kharis_desc:0 "Kharis is the delight, grace, and favor that is felt when gifts are returned when the gods and goddesses are honored. By cultivating Kharis with a patron deity the gods shall look happily upon us."
- doctrine_parameter_select_hellenic_god_active:1 "Rulers can [decide|E] to pick a patron avatar to venerate, gaining a related bonus"
- 
```
As you noticed, this is a conflict.

loup99 commented on README.md:
```diff
@@ -1 +1,4 @@
+# Butterfly-Project
+
+ Thanks to:
+ Chalk and Rootel ("More Faith Icons") for additional religion icons.
```
Did you get permission for that? If yes, I don't plan to have a Credits section in the readme, it will be on the forums.

LT-Rascek replied:
Yes, I inquired on Sept 6 on the mod's steam page:

Chalk  [author] Sep 6 @ 5:01pm 
@LT. Rascek yes, you can use this if you attribute me (and Rootel) :)

I put there for reference so I wouldn't forget.

loup99 replied:
Ok

loup99 replied:
You can remove this now, I have added it to the second post: https://forum.paradoxplaza.com/forum/threads/wip-a-migrational-era-mod-for-ck3.1414709/#post-26850337

LT-Rascek commented:
My largest question right now is whether or not to keep the Religion Doctrines or scrub them before the branch can be merged. I can fix the merge conflicts and localizations without trouble, but fleshing out the religion doctrines with specific effects will probably require more time, research, and work. Right now, the actual parameter effects and character modifiers are empty, but they can be written in greater detail.

Ideally for me this mod should be able to use religion council events to add/remove various ecumenism from Christian branches that are applied depending on the religious doctrines to more dynamically determine when Christian faiths are astray or hostile.

Other questions before revamping the pull request:
* Should I scrub the Cathars, Bogomils, Waldensians, and Lollards from the file, since they don't really occur until the 10th century (which I assume is outside the _initial_ scope of this mod). Or should they remain in?
* In a similar vein, do you want to add Aquelianism and Fraticelli into this mod as well? Or all the previous Christian faiths from WTWSMS?
* Should I clean up (add localizations and such) and add the religion council event so that others can take it and flesh it out in more detail?

loup99 commented:
> For completeness: I'll also note some specific Christian faiths from WTWSMS were not included on this pull request and instead modeled as Christian religious doctrines:
> Adoptionist, Antinomian, Apollinarian, Binitarian, Eutychian, Melchisedechian, Monarchian, Monophsyite, Monothelite, Pelagian, Pneumathomachian, Total Depravity,

I don't understand how you intend to represent them solely as doctrines. That means that there is no faith which has those doctrines at start, when several of those existed by 476 as beliefs and are present in counties, while others would appear later. 

> And some that weren't added don't have specific doctrines/tenets associated with them:
> Patripassian, Aquileian, Fraticelli, Psilantrhopist

Fraticelli isn't from WtWSMS, that was a vanilla heresy for catholicism. It is not a priority for us, I think we can safely ignore it for now. Aquileian on the other hand is important, it represents the schism of the Three Chapters, with a specific liturgical rite that was observed around the Patriarch of Aquileia. They were no longer in communion and would only reunite by 698. Patripassianism on the other hand is less important, it is a variant of Sabellianism, but it is nonetheless present at start in four counties. Psilanthropism is also a specific heresy which is not present on the map, could be a faith but not really important.

loup99 commented:
> My largest question right now is whether or not to keep the Religion Doctrines or scrub them before the branch can be merged. I can fix the merge conflicts and localizations without trouble, but fleshing out the religion doctrines with specific effects will probably require more time, research, and work. Right now, the actual parameter effects and character modifiers are empty, but they can be written in greater detail.

Does it really pose a problem to have doctrines without effects? In my opinion they have an added value if they describe doctrinal and theological differences, but they can not replace distinct faiths, just serve to define them.

> Ideally for me this mod should be able to use religion council events to add/remove various ecumenism from Christian branches that are applied depending on the religious doctrines to more dynamically determine when Christian faiths are astray or hostile.

Yes, although you might have to add "neutral" doctrines for matters which haven't yet been determined by council and would only later cause controversy. I don't know how that can work.

> Should I scrub the Cathars, Bogomils, Waldensians, and Lollards from the file, since they don't really occur until the 10th century (which I assume is outside the initial scope of this mod). Or should they remain in?

They should stay for compatibility with vanilla and future use, but never actually appear.

> In a similar vein, do you want to add Aquelianism and Fraticelli into this mod as well? Or all the previous Christian faiths from WTWSMS?

Fraticelli is not from WtWSMS, but for the others that were present on the map, yes.

LT-Rascek commented:
> Does it really pose a problem to have doctrines without effects? In my opinion they have an added value if they describe doctrinal and theological differences, but they can not replace distinct faiths, just serve to define them.

That's fine; I wasn't sure if you wanted to include them or not from your reply.

>Yes, although you might have to add "neutral" doctrines for matters which haven't yet been determined by council and would only later cause controversy. I don't know how that can work.

I'll be considering how to deal with that. I can change religion doctrines and tenets dynamically with events, so I'll be working on that as time goes on. For a first cut, we can have specific events relating to the Henotikon, Schism of the Three Chapters, and Monothelitism that will do that dynamically.

> They should stay for compatibility with vanilla and future use, but never actually appear.

I will reorganize the Christianity file accordingly.

> I don't understand how you intend to represent them solely as doctrines. That means that there is no faith which has those doctrines at start, when several of those existed by 476 as beliefs and are present in counties, while others would appear later.

Fair enough point; I'll focus for a first cut on porting all the Christian faiths from WtWSMS into this project, with a few exceptions I'll note below [1].

I'm looking into coding in dynamic "random" heresies so that these various doctrines don't have to be codified in a single religion, but instead could rise organically in the game. I'll shelve that in favor of porting in the other religions for the time being.

If I can make dynamic heresies work, I'll probably raise the issue again. Ideally, I want to cut down the number of faiths in the game if at all possible and instead let things arise randomly/organically, especially for faiths that are mostly centered on a single doctrine (Antinomianism, Total Depravity, Patripassianism, etc.) that don't have clear _mechanical_ differences from other faiths.

> Fraticelli was a vanilla heresy for Catholicism. It is not a priority for us, I think we can safely ignore it for now. Aquileian on the other hand is important, it represents the schism of the Three Chapters, with a specific liturgical rite that was observed around the Patriarch of Aquileia. They were no longer in communion and would only reunite by 698. Patripassianism on the other hand is less important, it is a variant of Sabellianism, but it is nonetheless present at start in four counties. Psilanthropism is also a specific heresy which is not present on the map, could be a faith but not really important.

As per above, I'll make the changes and include it in the next revision.

[1] - Exceptions:
* As you noted, merge Sabellianism with Patripassianism, since they are essentially the same view and faiths do not have to be heresies of just Orthodoxy or Catholicism, as was more-or-less the case with CK2.

* I'd consider merging adoptionism and psilantropist into Arianism in general.

LT-Rascek commented:
For completeness, I've implemented the following tenets with the following faiths listed. Effects are listed as below:

Universal Church - Chalcedon Christian
Can pick: Never
* Vulnerable to Hersey
* Minimum fervor of 30
* Same faith opinion +10, Realm Priest Opinion +10, Faith Creation Cost -25%
* When implemented, will have access to calling dynamic council events to modify the base faith

Marcionite Canon - Marcionism
Can Pick: If fundamentalist, if Gnostic, does not have Alexandrian Catechism, does not have Jewish Syncretic
Monthly Learning Lifestyle XP gain: +20%
[I'm open to suggestions for how to change this; I wasn't sure how best to represent it]

Rigorism - Donatism
Can Pick: Does not have Communion, Does have Fundamentalist
Considers other Christians Evil, Astray if they are Rigorist
Can excommunicate, but **no** indulgences
Considers Honest and Zealot Great Virtues, considers Deceitful and Cynical Great Sins
Voids Piety benefit of forgiving (I couldn't figure out how to remove forgiving as a virtue since it's a basic Christian virtue in game)

Ecstatic Prophecy - Montanism
Can Pick: Does not have Adoricism
Makes Possessed a Very Great Virture (+3 Piety)

Iconoclasm - Iconoclasts, Paulcians
Can Pick: Does not have Aniconism
Considers other Christians evil, other Iconoclasts astray (Probably should make Aniconism astray as well)
+10% piety, +10 same faith opinion

In keeping with history, perhaps Paulicians should loose Iconoclasm in favor of Aniconism if they get defeated in an uprising, as happened in history.

loup99 commented:
> I'll be considering how to deal with that. I can change religion doctrines and tenets dynamically with events, so I'll be working on that as time goes on. For a first cut, we can have specific events relating to the Henotikon, Schism of the Three Chapters, and Monothelitism that will do that dynamically.

Just to be clear since I realise my wording might open up to confusion: when I said "you" I didn't mean you in person or to give you any direction, more "one would have to". That being said, if you want to be in charge of this you are more than welcome to do it. I agree with the first specific events you suggest.

> Fair enough point; I'll focus for a first cut on porting all the Christian faiths from WtWSMS into this project, with a few exceptions I'll note below [1].
> 
> I'm looking into coding in dynamic "random" heresies so that these various doctrines don't have to be codified in a single religion, but instead could rise organically in the game. I'll shelve that in favor of porting in the other religions for the time being.
> 
> If I can make dynamic heresies work, I'll probably raise the issue again. Ideally, I want to cut down the number of faiths in the game if at all possible and instead let things arise randomly/organically, especially for faiths that are mostly centered on a single doctrine (Antinomianism, Total Depravity, Patripassianism, etc.) that don't have clear mechanical differences from other faiths.
> 

I fully agree with this for the obscure faiths that are not present at start, which would ideally be appearing dynamically and do not need to existed in an organised way at start. Amongst those you listed it applies for Antinomianism, Total Depravity and Psilanthropism, while the Patripassianism/Sabellianism merge should stay in if we keep it in counties. That being said, the Antinomians did have some unique things in CK2, like concubinage and female temple holders, but from a historical point of view there is not much that justify their specific inclusion. 

> [1] - Exceptions:
> 
> As you noted, merge Sabellianism with Patripassianism, since they are essentially the same view and faiths do not have to be heresies of just Orthodoxy or Catholicism, as was more-or-less the case with CK2.
> 

Yes, I think the name of the merge could be Patripassianism, since that was the religion we had on the map and which had a Patriarchy you could instate.

> I'd consider merging adoptionism and psilantropist into Arianism in general.

As I said above, Psilanthropism can be removed entirely/made into a dynamic heresy. For Adoptionism, they didn't exist at start so either it is saved as a future historical heresy, or it is incorporated dynamically?

LT-Rascek commented:
Applied the requested changes and rebased onto 339211f.
All Christian faiths from WtWSMS included, with the following exceptions:

Adoptionism - handled with Christian Doctrines now
Monothelitism - handled in the future with Christian doctrines
Psilantropist - handled with adoptionism doctrine
Sabellian - rolled into patripassianism
Fraticelli - outside initial mod scope

Also Sethianism, Priscinillinism, and Valentinianism are dualist faiths in CK3 and I don't see a reason to move them back into Christianity. Localization should be complete as well.

LT-Rascek commented:
> Just to be clear since I realise my wording might open up to confusion: when I said "you" I didn't mean you in person or to give you any direction, more "one would have to". That being said, if you want to be in charge of this you are more than welcome to do it. I agree with the first specific events you suggest.

For reference, I've started work on the Henotikon, but it is still a long way from complete. But I don't think it's required for this branch. When I have it written, you can decide if you want to use it or not; either way it's an educational process for me.

General effect is I envision is: Decreases Nicene Fervor yearly (in the absence of moral authority) and angers independent Nicene rulers/clergy in exchange for improved opinions of Nicene, Coptic, Apostolic vassals under your control and a bump to popular opinion (can't make that religion specific, unfortunately). Seems the closest to what occurred in WtWSMS.

> I fully agree with this for the obscure faiths that are not present at start, which would ideally be appearing dynamically and do not need to exist in an organised way at start. Amongst those you listed it applies for Antinomianism, Total Depravity and Psilanthropism, while the Patripassianism/Sabellianism merge should stay in if we keep it in counties. That being said, the Antinomians did have some unique things in CK2, like concubinage and female temple holders, but from a historical point of view there is not much that justify their specific inclusion.

Added the Antinomians in the commit fyi; pretty similar setup to Adamatism. I Think I missed equality doctrine though. If so, it's a one-line fix that I can take care of tomorrow.

loup99 commented:
> Applied the requested changes and rebased onto [339211f](https://github.com/loup99/BP/commit/339211ffe40e49a969783ff86eeb5dca3a80c07b).
> All Christian faiths from WtWSMS included, with the following exceptions:
> 
> Adoptionism - handled with Christian Doctrines now
> Monothelitism - handled in the future with Christian doctrines
> Psilantropist - handled with adoptionism doctrine
> Sabellian - rolled into patripassianism
> Fraticelli - outside initial mod scope

Ok, excellent. 

> Also Sethianism, Priscinillinism, and Valentinianism are dualist faiths in CK3 and I don't see a reason to move them back into Christianity. Localization should be complete as well.

Ok, but Sethianism, Priscillianism and Valentinianism should clearly be considered as heresies for Christians and be in conflict with Chalcedonian Christianity. I don't know how their addition by vanilla as Dualist will impact that, and if descriptions have to be altered to precise that it is Christian Sethianism, Priscillianism and Valentinianism.

> Lacking distinguishing characteristics, most of new faiths like Patripassianism and Total Depravity used the tenets Communion, Pentarchy, and Monasticism, since that seemed the most "default Christian" tenents.

Ok, Total Depravity wasn't necessary as a separate faith, but I can probably get back to you with some distinct tenets nonetheless.

> For reference, I've started work on the Henotikon, but it is still a long way from complete. But I don't think it's required for this branch. When I have it written, you can decide if you want to use it or not; either way it's an educational process for me.
> 
> General effect is I envision is: Decreases Nicene Fervor yearly (in the absence of moral authority in CK3) and angers independent Nicene rulers/clergy in exchange for improved opinions of Nicene, Coptic, Apostolic vassals under your control and a bump to popular opinion (can't make that religion specific, unfortunately). Seems the closest to what occurred in WtWSMS.

I agree with the general outline, and no, I don't think this branch needs to have the Henotikon included.

LT-Rascek commented:
Rebased onto 0bf71c4

> Ok, but Sethianism, Priscillianism and Valentinianism should clearly be considered as heresies for Christians and be in conflict with Chalcedonian Christianity. I don't know how their addition by vanilla as Dualist will impact that, and if descriptions have to be altered to precise that it is Christian Sethianism, Priscillianism and Valentinianism with the according clergy and hierarchy.

Heresies are controlled by events/religion_events/heresey_events.txt. So it is entirely possible to have outbreaks of dualist religions as a heresy; I've seen it happen in vanilla with strict heresies before. So separating them out like that doesn't seem to be a big issue in my book. As for terminology, its a few minor localization changes to ensure they use the same/similar terminology to Christianity with regards to clergy names/temple names/etc. I made similar changes to Ebionites, for example.

The separation between Gnostic Christian and Dualist faiths is, of course, a fuzzy concept. I'm of the opinion that at least Sethianism should be separated as Dualist, but I also think Priscillinsts and Valentinianism were different enough to have the distinction. 

As an aside, I've been thinking we'll need a Mazdan family with Dualism, Zoroastrianism, and Hepthalite religions, to keep it in line with WtWSMS. 

> Ok, Total Depravity wasn't necessary as a separate faith, but I can probably get back to you with some distinct tenets nonetheless.

After previous discussions, I decided to bring in the defaults from WtWSMS until such time as I can code a dynamic heresy mechanism. Seems the safest route to take at this time.

I think the branch is ready for merging, unless you want to add Sethianism/Priscillianism/Valentinianism to Christianity.

loup99 commented:
> Heresies are controlled by events/religion_events/heresey_events.txt. So it is entirely possible to have outbreaks of dualist religions as a heresy; I've seen it happen in vanilla with strict heresies before. So separating them out like that doesn't seem to be a big issue in my book. As for terminology, its a few minor localization changes to ensure they use the same/similar terminology to Christianity with regards to clergy names/temple names/etc. I made similar changes to Ebionites, for example.
> 
> The separation between Gnostic Christian and Dualist faiths is, of course, a fuzzy concept. I'm of the opinion that at least Sethianism should be separated as Dualist, but I also think Priscillinsts and Valentinianism were different enough to have the distinction.
> 
If in terms of mechanics there is strictly no difference with them being a Christian heresy, it can stay as it is.

> As an aside, I've been thinking we'll need a Mazdan family with Dualism, Zoroastrianism, and Hepthalite religions, to keep it in line with WtWSMS.

Good idea for Zoroastrian and Hephtalite, but taking out the Dualist religions from Abrahamic might pose a few problems on the other hand.

> After previous discussions, I decided to bring in the defaults from WtWSMS until such time as I can code a dynamic heresy mechanism. Seems the safest route to take at this time.

Ok.

LT-Rascek commented:
Rebased onto 5b3c5d8

Implemented Henotikon. Has the option to issue, option to revoke, and a yearly event that will void it if the Byzantine Empire disappears for some reason. Not certain under what circumstances it can be imposed on the broader Christian church. Will probably want to write that after the Kartevalian events are written.

LT-Rascek commented:
Found missing localizations; update and cleaned up.
